src/datachain/toolkit/split.py

```diff
@@ -58,14 +69,16 @@ def train_test_split(dc: DataChain, weights: list[float]) -> list[DataChain]:
 
     weights_normalized = [weight / sum(weights) for weight in weights]
 
-    resolution = 2**31 - 1  # Maximum positive value for a 32-bit signed integer.
+    rand_col = C("sys.rand")
+    if seed is not None:
+        uniform_seed = random.Random(seed).randrange(1, RESOLUTION)  # noqa: S311
```

Comment:
Good catch with the lower bound! I was relying on statistics and hoping it would never be `0` but better to make sure. 🙈 ❤️

Reply:
Note that here `RESOLUTION` should be `RESOLUTION + 1`[^1] because `randrange` behaves like range, and does not include the last element.

```suggestion
        uniform_seed = random.Random(seed).randrange(1, RESOLUTION + 1)  # noqa: S311
```

[^1]: Because you declared `RESOLUTION` as `2**31 - 1` above.